### ISSUE TITLE
__asm__ blocks indentation

### ISSUE THREAD

**Issue by mihaipopescu** [uncrustify/uncrustify#1236]
Asm blocks have a few ways of being opened and closed depending on the compiler. Detect and ignore the contents, including alignment.

Workaround: can always fall back on disable/enable_processing_cmt.

Expected versus output diff:
---
```diff
diff --git "a/C:\\work\\external\\uncrustify\\tests\\output\\staging\\10054-UNI-1344.cpp" "b/C:\\work\\external\\uncrustify\\tests\\results\\staging\\10054-UNI-1344.cpp"
index bda6487..f8af7aa 100644
--- "a/C:\\work\\external\\uncrustify\\tests\\output\\staging\\10054-UNI-1344.cpp"
+++ "b/C:\\work\\external\\uncrustify\\tests\\results\\staging\\10054-UNI-1344.cpp"
@@ -1,20 +1,20 @@
 // Asm blocks have their own special indentation where lables must remain at indent 0 relative to __asm__ block.
 // They few ways of being opened and closed depending on the compiler.
 // For now, we can at least detect and ignore the contents, including alignment.
 
 // Workaround: can always fall back on disable/enable_processing_cmt.
 
 void foo()
 {
     int head, bar;
     __asm__ __volatile__
     (
         "movq %0,%%xmm0\n\t"    /* asm template */
-    "0:\n\t"
-        "bar    %0, [%4]\n\t"    // in template
-    "1:\n\t"
+        "0:\n\t"
+        "bar\t%0, [%4]\n\t" // in template
+        "1:\n\t"
         : "=a", (bar)
         : "=&b", (&head), "+m", (bar)
         : "cc"
     );
 }
```

Debug support files:
---
[10054-UNI-1344.cpp](https://gist.githubusercontent.com/mihaipopescu/6a04c29dc9725a8465afb3ea59f76e4f/raw/bda64870a3d6e1cec07e2fa4606d8bd5094a0b94/10054-UNI-1344.cpp)
[10054-UNI-1344.cpp.log](https://gist.githubusercontent.com/mihaipopescu/6a04c29dc9725a8465afb3ea59f76e4f/raw/f7cf2084742709e5ee2484f19fb8a6b450d88e02/10054-UNI-1344.cpp.log)
[10054-UNI-1344.cpp.unc](https://gist.githubusercontent.com/mihaipopescu/6a04c29dc9725a8465afb3ea59f76e4f/raw/bb1f8f84a43d96842425b9c5b53d2f52f0b52a10/10054-UNI-1344.cpp.unc)
[Uncrustify.Cpp.cfg](https://gist.githubusercontent.com/mihaipopescu/6a04c29dc9725a8465afb3ea59f76e4f/raw/23d62d9dba5caf352fdc1793792e98eed4b81929/Uncrustify.Cpp.cfg)
[UNI-1344.cpp](https://gist.githubusercontent.com/mihaipopescu/6a04c29dc9725a8465afb3ea59f76e4f/raw/cd26ac1b9ad6d42f89b2e7a1c015f0e8bb734c92/UNI-1344.cpp)

Disclaimer:
---
Note: This is a generated issue from our internal issue tracker.
Expect a dry description which could be insufficient to properly understand the issue.
On the other hand, the support files should be sufficient to easily reproduce and debug the issue,
but please comment if it's still unclear and I will get back with clarifications.
See main discussion thread here #1223.

**Comment by guy-maurel** [comment]
Bugfix at #1502

**Comment by gmaurel** [comment]
Fixed at a31115470c41fb2ecbdf230a4a5e4d0669619a29, Dec 16 18:16:45 2017
